src/components/Book.js

```diff
@@ -15,6 +20,13 @@ Book.propTypes = {
   id: PropTypes.number.isRequired,
   title: PropTypes.string.isRequired,
   category: PropTypes.string.isRequired,
+  deleteBook: PropTypes.func.isRequired,
 };
 
-export default Book;
+const mapDispatchToProps = (dispatch) => {
+  const deleteBook = (id) => dispatch(deleteBookAction(id));
+
+  return { deleteBook };
+};
+
+export default connect(null, mapDispatchToProps)(Book);
```

Comment:
The Book component should not have access to redux. 
Add these lines of code to the bookList component with the necessary changes, then pass the function down as a prop and delete the lines of code from Book component.